apps/circular_dependencies.json

```diff
@@ -67,11 +67,12 @@
   "src/code-studio/components/SoundPicker.jsx -> src/code-studio/components/AssetManager.jsx -> src/clientApi.js -> src/code-studio/initApp/project.js -> src/code-studio/header.js -> src/code-studio/components/header/HeaderMiddle.jsx -> src/code-studio/components/header/ProjectInfo.jsx -> src/code-studio/components/header/ProjectBackedHeader.jsx -> src/code-studio/components/header/ProjectShare.jsx -> src/lab2/header/lab2HeaderShare.js -> src/lab2/views/Lab2ShareDialogWrapper.tsx -> src/code-studio/components/ShareDialog.jsx -> src/code-studio/components/ShareAllowedDialog.jsx -> src/code-studio/components/libraries/LibraryCreationDialog.jsx -> src/code-studio/components/libraries/libraryLoader.js -> src/code-studio/components/libraries/libraryParser.js -> src/lib/tools/jsinterpreter/JSInterpreter.js -> src/lib/tools/jsinterpreter/codegen.js -> src/dropletUtils.js -> src/StudioApp.js -> src/code-studio/assets.js -> src/code-studio/assets/show.js -> src/code-studio/components/SoundPicker.jsx",
   "src/code-studio/components/ShareDisallowedDialog.jsx -> src/code-studio/components/shareDialogRedux.js -> src/clientApi.js -> src/code-studio/initApp/project.js -> src/code-studio/header.js -> src/code-studio/components/header/HeaderMiddle.jsx -> src/code-studio/components/header/ProjectInfo.jsx -> src/code-studio/components/header/ProjectBackedHeader.jsx -> src/code-studio/components/header/ProjectShare.jsx -> src/lab2/header/lab2HeaderShare.js -> src/lab2/views/Lab2ShareDialogWrapper.tsx -> src/code-studio/components/ShareDialog.jsx -> src/code-studio/components/ShareDisallowedDialog.jsx",
   "src/code-studio/components/ShareDialog.jsx -> src/code-studio/components/ShareAllowedDialog.jsx -> src/templates/projects/publishDialog/PublishDialog.jsx -> src/templates/projects/publishDialog/publishDialogRedux.js -> src/clientApi.js -> src/code-studio/initApp/project.js -> src/code-studio/header.js -> src/code-studio/components/header/HeaderMiddle.jsx -> src/code-studio/components/header/ProjectInfo.jsx -> src/code-studio/components/header/ProjectBackedHeader.jsx -> src/code-studio/components/header/ProjectShare.jsx -> src/lab2/header/lab2HeaderShare.js -> src/lab2/views/Lab2ShareDialogWrapper.tsx -> src/code-studio/components/ShareDialog.jsx",
+  "src/templates/projects/submitProjectDialog/SubmitProjectDialog.tsx -> src/code-studio/components/shareDialogRedux.js -> src/clientApi.js -> src/code-studio/initApp/project.js -> src/code-studio/header.js -> src/code-studio/components/header/HeaderMiddle.jsx -> src/code-studio/components/header/ProjectInfo.jsx -> src/code-studio/components/header/ProjectBackedHeader.jsx -> src/code-studio/components/header/ProjectShare.jsx -> src/lab2/header/lab2HeaderShare.js -> src/lab2/views/Lab2ShareDialogWrapper.tsx -> src/templates/projects/submitProjectDialog/SubmitProjectDialog.tsx",
```

Comment:
Consulted with @thomasoniii on this. He said that we could try to add an extra file to try to break this circular dependency. However, if we see that there are no loading errors, then adding this exception is okay. I also see that there are other examples similar to this one involving `clientApi` and `header.js`.

Reply:
After a refactor, I can remove the exception! 🎉